src/runtime/recipe/tests/recipe-test.ts

```diff
@@ -873,6 +873,42 @@ describe('recipe', () => {
     assert.isTrue(particle.connections['a'].handle.capabilities.isEquivalent(
       Capabilities.create([new DeletePropagation(true)])));
   });
+  it('adds delete propagation capability to handles with hard refs in inline entity', async () => {
+    const recipe = (await Manifest.parse(`
+      schema Hr
+      schema Inner2
+        t: &Hr @hardRef
+      schema Inner
+        j: inline Inner2
+      particle MyParticle
+        a: writes Thing {i: inline Inner}
```

Comment:
what if `a: writes [Thing {i: inline Inner}]` - it's a collection, but we still want to find the annotation.

(as a side note, it might be too verbose to have a separate test case for each field type, so maybe combine them into one test case and the same recipe but with a handle per type)?

Reply:
added collection test case, and combined in a single test